src/ansys_sphinx_theme/__init__.py

```diff
@@ -107,6 +109,106 @@ def setup_default_html_theme_options(app):
     app.config.html_theme_options.setdefault("collapse_navigation", True)
 
 
+def fix_edit_html_page_context(
+    app: Sphinx, pagename: str, templatename: str, context: dict, doctree: document
+) -> None:
+    """Add a function that Jinja can access for returning an "edit this page" link .
+
+    This function creates an "edit this page" link for any library.
+    The link points to the corresponding file on the main branch.
+
+    Parameters
+    ----------
+    app : Sphinx
+        The Sphinx application instance for rendering the documentation.
+    pagename : str
+        The name of the current page.
+    templatename : str
+        The name of the template being used.
+    context : dict
+        The context dictionary for the page.
+    doctree : document
+        The document tree for the page.
+
+    Notes
+    -----
+    .. [1] Originally implemented by `Alex Kaszynski <https://github.com/akaszynski>`_ in
+    `PyVista <https://github.com/pyvista/pyvista>`_,
+    see https://github.com/pyvista/pyvista/pull/4113
+    """
+
+    def fix_edit_link_button(link: str) -> str:
+        """Transform "edit on GitHub" links to the correct URL.
+
+        This function fixes the URL for the "edit this page" link.
+
+        Parameters
+        ----------
+        link : str
+            The link to the GitHub edit interface.
+
+        Returns
+        -------
+        str
+            The link to the corresponding file on the main branch.
+        """
+        github_user = context.get("github_user", "")
+        github_repo = context.get("github_repo", "")
+        github_source = context.get("source_path", "")
+        kind = context.get("github_version", "")
+
+        if pagename.startswith("examples") and "index" not in pagename:
+            return f"http://github.com/{github_user}/{github_repo}/edit/{kind}/{pagename}.py"
+
```

Comment:
We are assuming ``.py`` files for the examples... but this might not always be the case. Take PyGeometry for example. We are using ``.mystnb``. I'd like this to be configurable somehow, or inferred..

Reply:
In that case, i will drop this for extension on examples page and we can improve this after doing the text cases in different libraries, if that works for you.